src/Neo/Ledger/TransactionVerificationContext.cs

```diff
@@ -50,14 +50,19 @@ public void AddTransaction(Transaction tx)
         /// Determine whether the specified <see cref="Transaction"/> conflicts with other transactions.
         /// </summary>
         /// <param name="tx">The specified <see cref="Transaction"/>.</param>
+        /// <param name="conflictingTxs">The list of <see cref="Transaction"/> that conflicts with the specified one and are to be removed from the pool.</param>
         /// <param name="snapshot">The snapshot used to verify the <see cref="Transaction"/>.</param>
         /// <returns><see langword="true"/> if the <see cref="Transaction"/> passes the check; otherwise, <see langword="false"/>.</returns>
-        public bool CheckTransaction(Transaction tx, DataCache snapshot)
+        public bool CheckTransaction(Transaction tx, IEnumerable<Transaction> conflictingTxs, DataCache snapshot)
         {
             BigInteger balance = NativeContract.GAS.BalanceOf(snapshot, tx.Sender);
             senderFee.TryGetValue(tx.Sender, out var totalSenderFeeFromPool);
 
             BigInteger fee = tx.SystemFee + tx.NetworkFee + totalSenderFeeFromPool;
+            foreach (var conflictTx in conflictingTxs)
+            {
+                fee -= conflictTx.NetworkFee + conflictTx.SystemFee;
```

Comment:
Is not balance instead of fee?

Reply:
The `balance` is from the snapshot of the verified, thus it is already updated, right?
Thus, as @shargon highlighted maybe it would be done by increasing balance?

Reply:
I don't understand why we substract the fee from the current tx. It could be negative.

Reply:
Maybe it is incorrect as you mentioned.

Reply:
What we're trying to do here is evaluate `balance`/`fee` relation if we're to replace conflicting transactions with a new one. So that's basically `totalSenderFeeFromPool + newTxFee - sum(conflictingTxFees...)`. This supposes that `totalSenderFeeFromPool` already includes `sum(conflicting...)`, but at the same time we need to check that we have the same sender in `conflictingTxs`, @AnnaShaleva?

Reply:
> but at the same time we need to check that we have the same sender in `conflictingTxs`

Exactly, and this part is missing from the PR, it's a mistake. I'll fix it.

Reply:
Fixed.

Reply:
Add a parentheses to make it clear `fee -= (conflictTx.NetworkFee + conflictTx.SystemFee)`

Reply:
I think that I understood the logic, looks good. But maybe change some variables names.

Reply:
Fixed!